src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -303,6 +303,39 @@ public StudentExamResource(ExamAccessService examAccessService, StudentExamServi
         return ResponseEntity.ok(testRun);
     }
 
+    /**
+     * GET /courses/{courseId}/exams/{examId}/student-exams/exercise-ids : Find the exercise ids for the current userexam.
+     * Currently this is used for exam notification problem statement updates as an auxiliary method.
+     *
+     * @param courseId  the course to which the student exam belongs to
+     * @param examId    the exam to which the student exam belongs to
+     * @return the ResponseEntity with status 200 (OK) and with the found exercise ids as body
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/student-exams/exercise-ids")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<Long[]> getStudentExamExerciseIds(@PathVariable Long courseId, @PathVariable Long examId) {
+        long start = System.currentTimeMillis();
+        User user = userRepository.getUserWithGroupsAndAuthorities();
+        log.debug("REST request to get the student exam exercise ids of user {} for exam {}", user.getLogin(), examId);
+
+        // 1st: load the studentExam with all associated exercises
+        Optional<StudentExam> optionalStudentExam = studentExamRepository.findWithExercisesByUserIdAndExamId(user.getId(), examId);
+        if (optionalStudentExam.isEmpty()) {
+            return notFound();
+        }
+        var studentExam = optionalStudentExam.get();
+
+        Optional<ResponseEntity<StudentExam>> courseAndExamAccessFailure = studentExamAccessService.checkCourseAndExamAccess(courseId, examId, user, studentExam.isTestRun());
+        if (courseAndExamAccessFailure.isPresent()) {
+            return null;
```

Comment:
`return null`
We rarely return null in Rest calls for failure states. At least I couldn't find another or frequent instances where we return null to the caller. What about returning `notFound()` here as above.

Reply:
you can use orElseThrow in here. EntityNotFoundException would be useful.

Reply:
Good point. Because this is a failure that occurs if the security check fails after studentExamAccessService.checkCourseAndExamAccess has been called I decided to use forbidden instead of notFound as the return value.